src/database-controller/src/core/k8s.js

```diff
@@ -0,0 +1,250 @@
+  const cluster = { name: 'cluster', server: config.customK8sApiServerURL };
+  const user = { name: 'user' };
+  kc.loadFromClusterAndUser(cluster, user);
+}
+
+// If API server reports a non-200 status code, the client will
+// throw an error. In such case, error.name will be "HttpError",
+// and error.response.statusCode is the actual status code.
+// If network is disconnected, the error will be a different one,
+// and you cannot get error.name and error.statusCode.
+const customObjectsClient = kc.makeApiClient(k8s.CustomObjectsApi);
+
+async function getFramework(name, namespace = 'default') {
+  const res = await customObjectsClient.getNamespacedCustomObject(
+    'frameworkcontroller.microsoft.com',
+    'v1',
+    namespace,
+    'frameworks',
+    name,
+  );
+  return res.response;
+}
+
+async function listFramework(name, namespace = 'default') {
+  const res = await customObjectsClient.listNamespacedCustomObject(
+    'frameworkcontroller.microsoft.com',
+    'v1',
+    namespace,
+    'frameworks',
+  );
+  return res.response;
+}
+
+async function createFramework(frameworkDescription, namespace = 'default') {
+  const res = await customObjectsClient.createNamespacedCustomObject(
+    'frameworkcontroller.microsoft.com',
+    'v1',
+    namespace,
+    'frameworks',
+    frameworkDescription,
+  );
+  return res.response;
+}
+
+async function patchFramework(name, data, namespace = 'default') {
+  const res = await customObjectsClient.patchNamespacedCustomObject(
+    'frameworkcontroller.microsoft.com',
+    'v1',
+    namespace,
+    'frameworks',
+    name,
+    data,
+    { headers: { 'Content-Type': 'application/merge-patch+json' } },
+  );
+  return res.response;
+}
+
+async function deleteFramework(name, namespace = 'default') {
+  const res = await customObjectsClient.deleteNamespacedCustomObject(
+    'frameworkcontroller.microsoft.com',
+    'v1',
+    namespace,
+    'frameworks',
+    name,
+  );
+  return res.response;
+}
+
+function getFrameworkInformer(
+  timeoutSeconds = 365 * 86400,
+  namespace = 'default',
+) {
+  /*
+  Usage:
+
+    const informer = getFrameworkInformer()
+    informer.on('add', (obj) => { console.log(`Added: ${obj.metadata.name}`); });
+    informer.on('update', (obj) => { console.log(`Updated: ${obj.metadata.name}`); });
+    informer.on('delete', (obj) => { console.log(`Deleted: ${obj.metadata.name}`); });
+    informer.on('error', (err) => { console.error(err);});
+    informer.start();
+
+  If the informer disconnects normally from API server, it will re-connect automatically.
+  But during the reconnection, listFn will be called again, which is inefficient.
+  According to https://github.com/kubernetes-client/javascript/blob/932c2fbc34db954c6ed397b3cd9ead08b2ff1d10/src/cache.ts#L82-L85,
+  this behavior will be fixed in the future.
+  TO DO: If @kubernetes/client-node fixes this issue, we should upgrade our code to use the new code.
```

Comment:
Comment we set a large timeoutSeconds to mitigate this

Reply:
Done.